src/OMSimulatorLib/ComponentFMUCS.cpp

```diff
@@ -1146,6 +1146,110 @@ oms_status_enu_t oms::ComponentFMUCS::getReal(const ComRef& cref, double& value)
   return getReal(vr, value);
 }
 
+
+oms_status_enu_t oms::ComponentFMUCS::getDirectionalDerivative(const ComRef& cref, double& value)
+{
+  if (!getModel().validState(oms_modelState_instantiated|oms_modelState_initialization|oms_modelState_simulation))
+    return logError_ModelInWrongState(getModel().getCref());
+
+  if (!getFMUInfo()->getProvidesDirectionalDerivative())
+    return logError("It is not possible to compute directionalDerivative for signal \"" + std::string(getFullCref() + cref) + "\" as providesDirectionalDerivative is false or not provieded in modelDescription.xml");
+
+  int j = -1;
+  for (size_t i = 0; i < allVariables.size(); i++)
+  {
+    if (allVariables[i].getCref() == cref && allVariables[i].isTypeReal())
+    {
+      j = i;
+      break;
+    }
+  }
+
+  if (!fmu || j < 0)
+    return logError_UnknownSignal(getFullCref() + cref);
+
+  if (oms_modelState_instantiated == getModel().getModelState())
+  {
+    if (getFMUInfo()->getGenerationTool().substr(0, 12) == "OpenModelica")
+      logWarning("It is not possible to get partial derivatives of OpenModelica generated fmus during initialization mode, getting directional derivative after intialization is possible");
```

Comment:
I expected this to become a warning within the fmu and not OMSimulator. But doesn't matter I guess, since we are going to implement support for it ASAP anyways :)